### PR TITLE
Add SwiftUI support to default assets resource synthesizer

### PR DESCRIPTION
**Pull request by kyungpyoda**
Resolves https://github.com/tuist/tuist/issues/3595

### Short description 📝

Update AssetsTemplate to support SwiftUI

> Ref
> [SwiftGen](https://github.com/SwiftGen/SwiftGen/blob/stable/SwiftGen.playground/Pages/XCAssets-Demo.xcplaygroundpage/Sources/Implementation%20Details.swift)

### CODE REVIEW

**Review by danyf90 — CHANGES_REQUESTED** [review]

**Review by luispadron — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Next
+
+### Added
+
+- Update AssetsTemplate to support SwiftUI [#4838](https://github.com/tuist/tuist/pull/4838) by [kyungpyoda](https://github.com/kyungpyoda).
```
[the remark this answers is not in the document]
Comment (kyungpyoda) [reply CHANGELOG.md]: I've deleted that commit. Thx :)

**Review by danyf90 — APPROVED** [review]

**Review by fortmarek — APPROVED** [review]

**Comment by kyungpyoda** [comment]
I really don't know what is wrong 🤔

**Comment by danyf90** [comment]
@kyungpyoda have you tried to check what is failing? keeping to merge main won't fix the issue :D

**Comment by danyf90** [comment]
@allcontributors add @kyungpyoda for code

**Comment by kyungpyoda** [comment]
@danyf90  I forgot a `#endif`... 😓 now I fixed it.

**Pull request merged**